packages/manager/modules/pci/src/projects/project/ai/apps/add/add.controller.js

```diff
@@ -262,27 +269,57 @@ export default class AppAddController {
         );
 
         presets.map((preset) => {
-          const partnerPrice = this.AppService.getPartnerPrice(
+          const partnerPriceCPU = this.AppService.getPartnerPrice(
             this.prices,
-            preset.partner.id,
-            preset.partner.flavor,
+            preset.partnerId,
+            preset.id,
             'cpu',
           );
+          const partnerPriceGPU = this.AppService.getPartnerPrice(
+            this.prices,
+            preset.partnerId,
+            preset.id,
+            'gpu',
+          );
+          const partner = partners.find((part) => part.id === preset.partnerId);
 
           return merge(preset, {
-            tax: partnerPrice.tax + resourcePrice.tax,
-            priceInUcents:
-              partnerPrice.priceInUcents + resourcePrice.priceInUcents,
+            partner,
+            prices: {
+              cpu: partnerPriceCPU,
+              gpu: partnerPriceGPU,
+            },
           });
         });
```

Comment:
**question:** `presets.map()` returns a new array but it is not assigned to any variable. So, I think `map` is not the appropriate function which can be used here.
If the intention is to modify the `presets` array itself, then may be we can reassign the returned array to `presets`

Reply:
Good point, changed here: https://github.com/ovh/manager/pull/9413/commits/8f753a833c9a5a9857c174fe32824151271b5412